test/report_aggregates_only_test.cc

```diff
@@ -0,0 +1,67 @@
+
+#undef NDEBUG
+#include <cstdio>
+#include <fstream>
+#include <streambuf>
+#include <string>
+#include <vector>
+
+#include "benchmark/benchmark.h"
+#include "output_test.h"
+
+// Ok this test is super ugly. We want to check what happens with the file
+// reporter in the presence of ReportAggregatesOnly().
+// We do not care about console output, the normal tests check that already.
+
+void BM_SummaryRepeat(benchmark::State& state) {
+  for (auto _ : state) {
+  }
+}
+BENCHMARK(BM_SummaryRepeat)->Repetitions(3)->ReportAggregatesOnly();
+
+int SubstrCnt(const std::string& haystack, const std::string& pat) {
+  if (pat.length() == 0) return 0;
+  int count = 0;
+  for (size_t offset = haystack.find(pat); offset != std::string::npos;
+       offset = haystack.find(pat, offset + pat.length()))
+    ++count;
+  return count;
+}
+
+int main(int argc, char* argv[]) {
```

Comment:
Is it worth putting some of this logic in the OutputTestHelper or a new helper that can be shared between the two tests?

Reply:
Hmm, maybe i can wrap some of this support code into RAII class..

Reply:
Yep, looks better now.